azure/pf-azure-context/classes/pf-azure-context.ps1

```diff
@@ -1,10 +1,32 @@
-
 Class PfAzureContext : PfCloudContext{
     [string]$SubscriptionName
     [string]$ResourceGroupName
+    [string]$Region
+    [string]$TeamAbbreviation
+    [string]$CompanyAbbreviation
+    [string]$GroupAbbreviation
+    [string]$EnvironmentLetter
 
-    PfAzureContext([string]$SubscriptionName,[string]$ResourceGroupName){
+    PfAzureContext([string]$SubscriptionName,[string]$ResourceGroupName,[string]$Region){
+        $this.SubscriptionName=$SubscriptionName
+        $this.ResourceGroupName=$ResourceGroupName
+        $this.Region=$Region
+    }
+    PfAzureContext([string]$CompanyAbbreviation,[string]$GroupAbbreviation,[string]$TeamAbbreviation,[string]$EnvironmentLetter,[string]$Region){
+        $this.CompanyAbbreviation=$CompanyAbbreviation
+        $this.GroupAbbreviation=$GroupAbbreviation
+        $this.EnvironmentLetter=$EnvironmentLetter
+        $this.TeamAbbreviation=$TeamAbbreviation
+        $this.Region=$Region
+    }
+    PfAzureContext([string]$SubscriptionName,[string]$ResourceGroupName,[string]$CompanyAbbreviation,[string]$GroupAbbreviation,[string]$TeamAbbreviation,[string]$EnvironmentLetter,[string]$Region){
         $this.SubscriptionName=$SubscriptionName
         $this.ResourceGroupName=$ResourceGroupName
+        $this.CompanyAbbreviation=$CompanyAbbreviation
+        $this.GroupAbbreviation=$GroupAbbreviation
+        $this.EnvironmentLetter=$EnvironmentLetter
+        $this.TeamAbbreviation=$TeamAbbreviation
+        $this.Region=$Region
     }
```

Comment:
PfAzureContext now drives pretty much all info needed to deploy a resource.  You really have 3 options
-Supply RGName, SubName, and Region for custom deployments
-Suplly companyabbr, teamabbr, groupabbr, envLetter, and region to have powerform create default naming standards and enterprise standards for you
-Don't use a PFAzureContext at all and do a complete custom deployment for each resource by settings the options property off the build object.

Reply:
where do Sub + RG names get populated in 2nd constructor?  Is it assuming you want to set that on each resource directly or that it will use the info it has to create the names (in which case does this happen on each resource definition when created?)

Reply:
It will create the names based on the info.  You would only use the 2nd constructor if you want to use default naming standards.  The names will get created on instantiation of each resource type.